File: JobApplicationTracker.Frontend/types/Dashboard/DashboardStatsDto.ts
export interface DashboardStatsDto {
  totalApplications: number;
  activeApplications: number;
  interviewsScheduled: number;
  offersReceived: number;
  applicationsByStatus: Record<number, number>;
  applicationsThisWeek: number;
  applicationsThisMonth: number;
}

File: JobApplicationTracker.Api/DTOs/Dashboard/DashboardStatsDto.cs
namespace JobApplicationTracker.DTOs;

public class DashboardStatsDto
{
    public int TotalApplications { get; set; }
    public int ActiveApplications { get; set; }
    public int InterviewsScheduled { get; set; }
    public int OffersReceived { get; set; }
    public Dictionary<int, int> ApplicationsByStatus { get; set; } = new();
    public int ApplicationsThisWeek { get; set; }
    public int ApplicationsThisMonth { get; set; }
}
